core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php

```diff
@@ -54,6 +55,7 @@ protected function createAuthProvider(ContainerBuilder $container, $id, $config,
             ->setDefinition($twoFactorProviderId, new ChildDefinition(TwoFactorFactory::PROVIDER_DEFINITION_ID))
             ->replaceArgument(0, $id)
             ->replaceArgument(1, [])
+            ->replaceArgument(3, new Reference(BackupCodeManager::class))
```

Comment:
I don't know if this will be an issue, but you can't provide your own implementation of the backup code manager anymore. Before this, it was a configurable service.

Reply:
Imho you can, either by creating a compiler pass that overrides it, or even better by decorating the Contao service.

Reply:
Because of the deadline for the RC phase, I am going to merge this PR even though this discussion is unresolved. We still have until February 15th to fix this if we decide that the service should be configurable without decorating the service.

Reply:
Got it, all good 👍